member-service/src/main/java/com/example/memberservice/auth/token/service/AuthService.java

```diff
@@ -36,12 +37,17 @@ public TokensOutput reissueAccessTokenByRefreshToken(String refreshToken) {
 
         refreshTokenRepository.saveRefreshToken(memberCode, newRefreshToken);
 
-        boolean isSignedUp = memberJpaRepository.existsByCode(memberCode);
+        Optional<Members> optionalMembers = memberJpaRepository.findByCode(memberCode);
+        
 
-        String newAccessToken = jwtTokenGenerator.generateAccessToken(memberCode, isSignedUp);
+        Members members = optionalMembers.orElse(null);
+        String newAccessToken = (members != null)
+            //회원 가입 한 사용자의 경우 memberRole을 claims 에 추가
+            ? jwtTokenGenerator.generateAccessToken(members.getCode(), true, members.getRole())
+            //회원 가입 안한 사용자의 경우 memberRole을 claims 에 제거
+            : jwtTokenGenerator.generateAccessToken(memberCode, false);
```

Comment:
아래와 같이 작성하면 null을 직접 다루지 않아도 될 것 같아요!

```suggestion
        String newAccessToken = optionalMembers
                .map(members -> // Optional 안에 값이 존재할 때 실행할 로직
                    jwtTokenGenerator.generateAccessToken(members.getCode(), true, members.getRole()))
                .orElseGet(() -> // Optional 안에 값이 존재하지 않을 때 실행할 로직
                    jwtTokenGenerator.generateAccessToken(memberCode, false));
```

Reply:
stream 에 익숙해져야할 텐데 

아직까지도 새로운 문법들을 알게 됩니다. 적용 시켜 보겠습니다!

Reply:
최신 Refactor PR 에 반영하였습니다.